libpod/stats_freebsd.go

```diff
@@ -0,0 +1,174 @@
+package libpod
+
+import (
+	"math"
+	"strings"
+	"time"
+
+	"github.com/containers/common/pkg/cgroups"
+	"github.com/containers/podman/v4/libpod/define"
+	"github.com/containers/podman/v4/pkg/rctl"
+	"github.com/containers/storage/pkg/system"
+	"github.com/sirupsen/logrus"
+)
+
+// GetContainerStats gets the running stats for a given container.
+// The previousStats is used to correctly calculate cpu percentages. You
+// should pass nil if there is no previous stat for this container.
+func (c *Container) GetContainerStats(previousStats *define.ContainerStats) (*define.ContainerStats, error) {
```

Comment:
Any chance you can refactor the basic skeleton (line 19 to around line 42, I would think?) to be shared between Linux and FreeBSD? I don't think it's likely any of this changes, but in the off chance we do need to make stats run for paused containers or something similar, I'd like to not have to make the changes in two places.

Reply:
Good idea - I have split GetContainerStats into command and platform parts.